src/bit-systems/network-debug.ts

```diff
@@ -0,0 +1,47 @@
+import { defineQuery, enterQuery, exitQuery, removeEntity } from "bitecs";
+import { Material, Scene, Vector3 } from "three";
+import { Text as TroikaText } from "troika-three-text";
+import { HubsWorld } from "../app";
+import { Networked } from "../bit-components";
+import { NetworkDebugPrefab } from "../prefabs/network-debug";
+import { formatComponentProps, formatObjectName } from "../react-components/debug-panel/ECSSidebar";
+import { renderAsEntity } from "../utils/jsx-entity";
+import { forEachMaterial } from "../utils/material-utils";
+import { EntityID } from "../utils/networking-types";
+
+const texts = new Map<EntityID, TroikaText>();
```

Comment:
System design question:

Is it a good system design that a system has map from entity id to data? I thought in ECS a system shouldn't manage data by itself and all data should be represented as components.

/cc @netpro2k

Reply:
You're probably right about that, though I don't feel strongly about it one way or another.

Here we are mapping from `EntityID` of the `Networked` entity to the text3D object of the entity with the `NetworkedDebug` component. If we want to manage this state using bitecs, we would add another component (e.g. `NetworkedDebugText` with a single property, which would be the a reference to the `NetworkedDebug` entity:

```ts
export const NetworkDebugText = defineComponent({
  targetRef: Types.eid
})
```

Then, when we render the entity using the `NetworkDebugPrefab`, we'd add the `NetworkDebugText` to the `NetworkedEntity` and set its `targetRef` to the `NetworkDebugPrefab`'s `eid`:

```diff
-      const textObj = world.eid2obj.get(renderAsEntity(world, NetworkDebugPrefab()))! as TroikaText;
-      texts.set(eid, textObj);
+      const debugEid = renderAsEntity(world, NetworkDebugPrefab());
+      addComponent(world, NetworkedDebugText, eid);
+      NetworkedDebugText.targetRef[eid] = debugEid;
+      const textObj = world.eid2obj.get(debugEid)! as TroikaText;
```

Similarly, in the `exitedQuery` we get the eid from the component store:
```diff
   exitedNetworkedQuery(world).forEach(eid => {
-    removeEntity(world, texts.get(eid)!.eid!);
-    texts.delete(eid);
+    const debugEid = NetworkedDebugText.targetRef[eid];
+    removeEntity(world, debugEid);
   });
   ```
   
   And where we are doing the updating, we just use the ref to access the object3D:
   
   ```diff
     networkedQuery(world).forEach(eid => {
     const networkedObj = world.eid2obj.get(eid)!;
-    const textObj = texts.get(eid)!;
+    const textObj = world.eid2obj.get(NetworkedDebugText.targetRef[eid])! as TroikaText;
```

If we like this better then I'll commit these changes and remove the map. 

In practice, the map will be smaller than the component store, but if we're hesitant to add components to store data then we need to solve that problem (e.g. by archetypes). This is something that Dom and I have discussed but have not yet explored.

Reply:
Updated in https://github.com/mozilla/hubs/pull/5842/commits/38f6d6b89d770eb7645f8520736eb1ec22b5b2be